stdlib/src/utils/string_slice.mojo

```diff
@@ -31,6 +31,33 @@ alias StaticString = StringSlice[StaticConstantLifetime]
 """An immutable static string slice."""
 
 
+fn _count_utf8_continuation_bytes(span: Span[UInt8]) -> Int:
+    alias sizes = (256, 128, 64, 32, 16, 8)
+    var ptr = span.unsafe_ptr()
+    var num_bytes = len(span)
+    var amnt: Int = 0
+    var processed = 0
+
+    @parameter
+    for i in range(len(sizes)):
+        alias s = sizes.get[i, Int]()
+
+        @parameter
+        if simdwidthof[DType.uint8]() >= s:
+            var rest = num_bytes - processed
+            for _ in range(rest // s):
+                var vec = (ptr + processed).load[DType.uint8, width=s]()
+                var comp = (vec & 0b1100_0000) == 0b1000_0000
+                amnt += int(comp.cast[DType.uint8]().reduce_add())
+                processed += s
+
+    for i in range(num_bytes - processed):
+        amnt += int((ptr[processed + i] & 0b1100_0000) == 0b1000_0000)
+
+    _ = span
```

Comment:
I think this is needed.

Reply:
yes it is ?

Reply:
Sorry, typo.

Reply:
do you mean the for loop for the remaining bytes or the span explicit discard?

- for the for loop my thoughts were: 
The other option is to build a `SIMD[DType.uint8, 8]` and progressively load the pointer, and then do the `&`, `!=`, `reduce_and()`. This could be wasted operations vs just processing the remaining slots. I've seen in some algorithms for things like these (using SIMD) that the performance for the last < 8 bytes are sometimes just better by doing index access than vector ops. Might be wrong here since I haven't really measured the perf. diff. (since here we do 2 operations not just 1 so there might be wasted cycles or the pipeline scheduler could do it's job well :man_shrugging: )

- for the explicit discard of span:
the lifetime of the span outside the function is unknown and the pointer is used inside this func, so I'd rather ensure the owner of the data doesn't dealloc the pointer before the func has counted everything. Again, might be wrong since I don't understand lifetimes all that well.

WDYT?

Reply:
I meant the `_ = span`. It will be kept alive for the duration of the function.

Reply:
But isn't that because it has `ImmutableLifetime`? I also see that in all cases some indexing of the span is still done, which besides hurting performance because of index normalization it also means that the lifetime is ensured (?). 

PS: I'll delete the discard, but I still want to understand

Reply:
A borrowed argument (unlike a `var`) is kept alive for the function call. See also `Formatter.write_str` and `_quicksort_partition_right`.